### PR TITLE
MNT: consolidate exceptions and some relative imports

### LINKED ISSUE #125: Move all exception classes to a shared location and consolidate
## Current Behavior

[pcdsdaq/exceptions.py
](https://github.com/pcdshub/pcdsdaq/blob/cc9818ad3d3c0310e02e8925c4c611543fbca48d/pcdsdaq/exceptions.py#L1)[pcdsdaq/daq/interface.py](https://github.com/pcdshub/pcdsdaq/blob/cc9818ad3d3c0310e02e8925c4c611543fbca48d/pcdsdaq/daq/interface.py#L93)

### PR DESCRIPTION
**Pull request by klauer**
## Description
Move all daq exceptions into a central `pcdsdaq.exceptions` for easy importability/discoverability/etc.

## Motivation and Context
Closes #125 
Back-compat for exception imports was not retained as this is a newer structure
However this can be rectified easily if it seems worthwhile

## How Has This Been Tested?
Test suite on CI

## Where Has This Been Documented?

Issue and PR text

### CODE REVIEW

File: pcdsdaq/daq/__init__.py [hunk 1]
```diff
@@ -17,9 +17,14 @@
   lcls1-compatible interface for maximum parity between lcls1 and lcls2.
   It will be importable once it is ready for testing.
 """
-# flake8: noqa
-from .interface import (DaqError, DaqStateTransitionError, DaqTimeoutError,
-                        get_daq)
```
Comment (klauer) [inline pcdsdaq/daq/__init__.py:22]: This doesn't back-compat imports for exceptions here.
I doubt any code is relying on this, so that's probably OK?

Comment (tangkong) [reply pcdsdaq/daq/__init__.py:22]: A quick search through our org doesn't show any matches for these exceptions outside of this library, so I think we're good here

**Review by tangkong — APPROVED** [review]
Makes sense, looks clean.  LGTM

File: pcdsdaq/daq/__init__.py (hunk 1 above)
Comment (ZLLentz) [reply pcdsdaq/daq/__init__.py:22]: I think it's ok to not backport it, it's underutilized (maybe because it's buried)

**Review by ZLLentz — APPROVED** [review]
This looks clean and correct to me!

**Pull request merged**